lib/db_connection/connection.ex

```diff
@@ -355,6 +356,11 @@ defmodule DBConnection.Connection do
     end
   end
 
+  def handle_info({:timeout, timer, :trigger_hibernate},
+                  %{idle_hibernate: idle_hibernate} = s) when is_reference(timer) do
+    maybe_hibernate(idle_hibernate, {:noreply, %{s | hibernate_timer: nil}})
```

Comment:
We shouldn't hibernate here, first we need to call the current logic in `handle_info(:timeout, ..)` that follows this clause: https://github.com/elixir-ecto/db_connection/pull/103/files/5b3c7cbd2333f15b8ca30bd06bcd9533820c2db0..e1590494cd81e376261aa7d9592a709c13028e85#diff-794f06ecb2c5f02e8a11e735c3ee2339R364